### PR TITLE
Turn schema field in operations into a pinned relation

### LINKED ISSUE #245: Make `SchemaId::Application` value a `PinnedRelation` type
```yaml
action: update | create | delete
version: 1
schema: [<operation id>, ...] # This will be a pinned relation
fields: ...
```

### PR DESCRIPTION
**Pull request by adzialocha**
```yaml
action: update | create | delete
version: 1
schema: [<operation id>, ...] # This is a pinned relation now!
fields: ...
```

**Warning:** This obviously breaks _everything_ once again :smile_cat: 

**Fun fact:** I had to change the key pair in the `document` examples and documentation to preserve the ordering of operations as the entry hashes are different now

Closes #245 

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header
- [x] Check if descriptions and terminology match `handbook` content (and visa-versa) 
  - https://github.com/p2panda/handbook/pull/148

### CODE REVIEW

**Comment by sandreae** [comment]
Haha, nice fun fact :smile:

File: p2panda-js/src/session/session.ts [hunk 1]
```diff
@@ -82,11 +83,17 @@ export class Session {
...
-   * @param val schema hash
+   * @param val schema id
    * @returns Session
    */
-  setSchema(val: string): Session {
-    this._schema = val;
+  setSchema(val: SchemaId | string): Session {
+    if (typeof val === 'string') {
+      // Automatically convert to document view id when given only a hash
+      this._schema = [val];
```
Comment (sandreae) [inline p2panda-js/src/session/session.ts:92]: This is nice, as it will often be a single hash.

File: p2panda-rs/src/schema/schema_id.rs [hunk 2]
```diff
@@ -53,16 +59,62 @@ impl FromStr for SchemaId {
     }
 }
 
+/// Serde `Visitor` implementation used to deserialize `SchemaId`.
+struct SchemaIdVisitor;
+
+impl<'de> Visitor<'de> for SchemaIdVisitor {
```
Comment (sandreae) [inline p2panda-rs/src/schema/schema_id.rs:65]: Cool, I haven't seen this before!

**Review by sandreae — APPROVED** [review]
Looks good to me! We really are breaking everything often these days.....

File: p2panda-rs/src/schema/schema_id.rs (hunk 2 above)
Comment (adzialocha) [reply p2panda-rs/src/schema/schema_id.rs:65]: Me neither 🤣

**Pull request merged**